core/src/main/scala/chisel3/experimental/Defaulting.scala

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.experimental
+
+import chisel3._
+import chisel3.internal.sourceinfo.SourceInfo
+
+import scala.collection.immutable.SeqMap
+
+/** Data type for representing any type that has a defined default value
+  *
+  * Uses the underlying opaque type support.
+  *
+  * @note This API is experimental and subject to change
+  */
+final class Defaulting[T <: Data] private (
```

Comment:
should we delete this from this PR?

Reply:
Probably, but I want to try out this PR first to see how it works out.